File: c++/DegreeofAnArray.cpp
class Solution {
public:
    int findShortestSubArray(vector<int>& nums) {
        unordered_map<int, vector<int>> freqMap;
        int minWin = INT_MAX;
        int degree = 1;
        
        for (int inx = 0; inx < nums.size(); ++inx) {
            int key = nums[inx];
            
            if (freqMap.find(key) == freqMap.end()) {
                freqMap[key] = {1, inx, inx};
            } else {
                freqMap[key][0] += 1;
                freqMap[key][2] = inx;
                degree = max(degree, freqMap[key][0]);
            }
        }
        
        for (auto key : freqMap) {
            if (key.second[0] == degree) {
                minWin = min(minWin, key.second[2] - key.second[1] + 1);
            }
        }
        
        return minWin;
    }
};

File: python3/DegreeofAnArray.py
class Solution:
    def findShortestSubArray(self, nums: List[int]) -> int:
        degree = 1
        winMap = {}
        
        for index, val in enumerate(nums):
            if val in winMap:
                winMap[val][0] += 1
                winMap[val][2] = index
                degree = max(degree, winMap[val][0])
            else:
                winMap[val] = [1, index, index]
        
        minWin = float("inf")
        for key in winMap:
            if winMap[key][0] == degree:
                minWin = min(minWin, winMap[key][2] - winMap[key][1] + 1)
        
        return minWin
